File: src/java/BouncyNumber.java
package java;

public class BouncyNumber {

    public static boolean isBouncyNumber(long num) {
        char[] numStr = String.valueOf(num).toCharArray();
        boolean isIncreasing = false;
        boolean isDecreasing = false;

        for (int i = 0; i < numStr.length - 1; i++) {
            if (numStr[i] > numStr[i+1])
                isDecreasing = true;
            if (numStr[i] < numStr[i+1])
                isIncreasing = true;
        }

        return isDecreasing && isIncreasing;
    }

}

File: src/C++/BouncyNumber.cpp
#include <iostream>

using namespace std;

bool isBouncyNumber(long n);
bool isBouncyNumber(long n) {
    bool isdecreasing = false, isincreasing = false;
    int digits[20] = {0};
    for (int i = 0; i < 10; i++) {
        digits[i] = n % 10;
        n /= 10;
    }

    for (int i = 0; i < 10; i++) {
        if (digits[i + 1] < digits[i])
            isdecreasing = true;
        if (digits[i + 1] > digits[i])
            isincreasing = true;
    }
    return isincreasing && isdecreasing;
}